jbmc/unit/java_bytecode/java_replace_nondet/replace_nondet.cpp

```diff
@@ -54,11 +54,10 @@ void validate_nondet_method_removed(
 
     if(inst.is_return())
     {
-      const code_returnt &ret_expr = to_code_return(inst.code);
-      if(ret_expr.return_value().id() == ID_side_effect)
+      const auto &return_value = inst.return_value();
+      if(return_value.id() == ID_side_effect)
```

Comment:
Not for this PR: we need to re-consider whether "nondet" really should remain a "side effect." It'll eventually be the only side effect remaining in goto programs.

Reply:
Agreed.